### PR TITLE
Improved page 404 and implemented a different layout for it.

### PR DESCRIPTION
**Pull request by skkr**

### CODE REVIEW

**Review by AugustoL — COMMENTED** [review]
@skkr please rebase with current master and resolve conflicts

**Comment by skkr** [comment]
@AugustoL shoud be fine now.

File: src/layouts/LayoutError.js [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+
+  constructor(props) {
+    super(props);
+    this.state = {
+      name: 'MainLayout'
+    };
+  }
+
+  componentWillMount() {
+    const self = this;
```
Comment (MatiasOS) [inline src/layouts/LayoutError.js]: please, remove this assignment and  the few next

File: src/layouts/LayoutError.js [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+
+  componentDidMount() {
+    const self = this;
+  }
+
+  render() {
+    const self = this;
+
+    let childrenWithProps = React.cloneElement(self.props.children, {
+      _showAlert: (type, message) => self._alert.open(type, message)
```
Comment (MatiasOS) [inline src/layouts/LayoutError.js]: please, extract  anon function outside the render method

**Review by MatiasOS — CHANGES_REQUESTED** [review]

**Review by MatiasOS — APPROVED** [review]
:+1:

**Pull request merged**